YAXLib/Deserialization.cs

```diff
@@ -824,10 +824,28 @@ private bool TrySetValueForEmptyElement(object obj, MemberWrapper member, Type m
         // Get the container object from the element (may be null)
         _ = TryGetContainerObject(xElement, collType, memberAlias, out var containerObj);
 
-        var dataItems = new List<object?>(); // this will hold the actual data items
         var collItemType = ReflectionUtils.GetCollectionItemType(collType);
-        var isPrimitive = ReflectionUtils.IsBasicType(collItemType);
 
+        Type? dataItemsType = null;
+        IList dataItems; // this will hold the actual data items
+        if (collItemType == typeof(object))
+        {
+            dataItems = new List<object>();
+        }
+        else
+        {
+            try
+            {
+                dataItemsType = typeof(List<>).MakeGenericType(collItemType);
+                dataItems = (IList) Activator.CreateInstance(dataItemsType)!;
+            }
+            catch (Exception)
+            {
+                dataItems = new List<object>();
+            }
+        }
```

Comment:
In favor of cognitive complexity of method ` DeserializeCollectionValue` could this be wrapped into TryGet-alike method?